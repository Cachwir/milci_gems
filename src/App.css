.switch-wrapper {
  display: grid;
  grid-template-columns: 100%;
}

.switch-wrapper > div {
  grid-row-start: 1;
  grid-row-end: 1;
  grid-column-start: 1;
  grid-column-end: 1;
}

.container.floated {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}
.container .floated {
  float: none !important;
  display: block;
  position: relative !important;
}

@media only screen and (max-width: 767px){
  .about .container.floated {
    display: block;
  }
}
