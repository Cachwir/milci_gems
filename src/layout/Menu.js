import React, {Component} from "react";
import {Link, withRouter} from "react-router-dom";
import PropTypes from "prop-types";
import AppRouter from "../AppRouter";

class MenuWithoutRouter extends Component
{
    static propTypes = {
        location: PropTypes.object.isRequired,
        Parent: PropTypes.object.isRequired,
    };

    // pages' names come from AppRouter.ROUTES
    static PAGES_DATA = {
        HOME: {
            path: () => { return this.getPagePath(AppRouter.ROUTES.HOME)},
            icon: "home",
            text: "Home",
        },
        GALLERY: {
            path: () => { return this.getPagePath(AppRouter.ROUTES.GALLERY)},
            icon: "briefcase",
            text: "Galerie",
        },
        ABOUT: {
            path: () => { return this.getPagePath(AppRouter.ROUTES.ABOUT)},
            icon: "pencil",
            text: "Ã€ propos",
        },
    };

    static getPagePath(pageId) {
        return AppRouter.ROUTES_DATA[pageId].path;
    }

    Parent;

    constructor(props) {
        super(props);
        this.Parent = props.Parent;
        this.state = {currentPage: AppRouter.getRouteFromThisRouterPointOfView(this.props.location)};
        this.Parent.registerRouterEvent(this.onRouteChange)
    }

    onRouteChange = (appRouterRoute) => {
        this.setState({currentPage: appRouterRoute});
    };

    render() {
        return (
            <div>
                <nav id="navigation" className="style-1">
                    <div className="left-corner"></div>
                    <div className="right-corner"></div>
                    <ul className="menu" id="responsive">
                        {Object.keys(this.constructor.PAGES_DATA).map((pageKey, index) => {
                            const page = this.constructor.PAGES_DATA[pageKey];
                            const liId = this.state.currentPage === pageKey ? "current" : null;
                            return (
                                <li key={index}>
                                    <Link to={page.path()} id={liId}><i className={`halflings white ${page.icon}`}></i> {page.text}</Link>
                                </li>
                            );
                        })}
                    </ul>
                </nav>
                <div className="clearfix"></div>
            </div>
        );
    }
}

// Create a new component that is "connected" (to borrow redux
// terminology) to the router.
const Menu = withRouter(MenuWithoutRouter);

export default Menu;