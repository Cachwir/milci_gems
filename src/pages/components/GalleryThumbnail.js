import React from 'react';
import PropTypes from 'prop-types';
import {Link} from "react-router-dom";

const GalleryThumbnail = ({ url, path, title, subTitle, onThumbnailClick }) => (
    <div>
        <Link to={url} className="portfolio-item isotope" onClick={onThumbnailClick}>
            <figure>
                <div className="picture"><img src={path}
                                              alt=""/>
                    <div className="image-overlay-link"></div>
                </div>
                <figcaption className="item-description">
                    <h5>{title}</h5>
                    <span>{subTitle}</span>
                </figcaption>
            </figure>
        </Link>
    </div>
);

GalleryThumbnail.propTypes = {
    url: PropTypes.string.isRequired,
    path: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
};

export default GalleryThumbnail;