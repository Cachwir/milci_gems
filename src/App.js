import React, {Component} from "react";

import "./App.css";
import Header from "./layout/Header";
import Menu from "./layout/Menu";
import Footer from "./layout/Footer";
import AppRouter from "./AppRouter";
import {BrowserRouter as Router} from "react-router-dom";

class App extends Component {

  /**
   * L'appli sera composée de trois pages :
   * - une page de garde
   * - une gallerie
   * - à propos (pour expliquer comment ce site a été fait, dans quel contexte et ajouter des remerciements et des
   * sources)
   */

  routerEvents = [];

  registerRouterEvent(event)
  {
      this.routerEvents.push(event);
  }

  render() {
      return (
          <Router>
              <div id="wrapper">
                  <Header />
                  <Menu Parent={this} />
                  <AppRouter onRouteChangeEvents={this.routerEvents}  />
                  <Footer />
              </div>
          </Router>
      );
  }
}

export default App;
