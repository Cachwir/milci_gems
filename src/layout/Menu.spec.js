import { shallow } from 'enzyme'
import { expect } from 'chai'
import sinon from 'sinon'
import React from 'react'
import Menu from "./Menu";
import App from "../App";
import {MemoryRouter} from "react-router-dom";
import GalleryThumbnail from "../pages/components/GalleryThumbnail";

describe('<Menu />', () => {
    // it('contains all the links to the AppRouter\'s routes', () => {
    //     console.log(Object.keys(Menu.PAGES_DATA));
    //     const FalseApp = shallow(<App />);
    //     const wrapper = shallow(
    //         <MemoryRouter>
    //             <Menu Parent={FalseApp} />
    //         </MemoryRouter>
    //     );
    //     Object.keys(Menu.PAGES_DATA).forEach((pageDataKey) => {
    //         console.log(Menu.PAGES_DATA[pageDataKey].path());
    //         expect(wrapper.find("Link[to=\"/gallery\"]")).to.exist();
    //         // expect(wrapper.find("Link[to=\""+ Menu.PAGES_DATA[pageDataKey].path() +"\"]")).to.exist();
    //     });
    // });

    it('should match its reference snapshot', () => {
        const FalseApp = shallow(<App />);
        const wrapper = shallow(
            <Menu Parent={FalseApp} />
        );

        expect(wrapper).to.matchSnapshot();
    });

    // it('should trigger its `onClick` prop when clicked', () => {
    //     const onClick = sinon.spy();
    //     const wrapper = shallow(
    //         <Menu card="😁" feedback="hidden" index={0} onClick={onClick} />
    //     );
    //
    //     wrapper.simulate('click');
    //     expect(onClick).to.have.been.calledWith(0)
    // })
    //
    // it('should match its reference snapshot', () => {
    //     const onClick = sinon.spy();
    //     const wrapper = shallow(
    //         <Card card="😁" feedback="hidden" index={0} onClick={onClick} />
    //     );
    //
    //     expect(wrapper).to.matchSnapshot();
    // })
})