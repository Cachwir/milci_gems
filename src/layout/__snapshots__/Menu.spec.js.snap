// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should match its reference snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
